### ISSUE TITLE
Case sensitivity in spelling.txt and other files - 2023-04-14

### ISSUE THREAD

**Issue by marcoagpinto** [languagetool-org/languagetool#8258]
Hello!

I am opening a new ticket just for this.

I was wondering if Hunspell behaviour could be mimicked into spelling.txt and the other related files?

Hunspell works like this:
If a word is inserted in the dictionary in lowercase, a user can type it however he wishes.

If a word in the dictionary has one or more letters in uppercase, it needs to be typed exactly like it is in the dictionary, being the exception if it is all written by the user ALL in uppercase.

I was wondering if this could be implemented?

Basically, here is the pseudocode for the algorithm for words in spelling.txt, for example:
`If (current_word$) != LowerCase(current_word$) : activate_case_sensitivity_for_current_word=#True : EndIf`

Basically, this summarises it all.

Thanks!

**Comment by baarsrj** [comment]
But.... Hunspell does not support word groups, spelling,txt does. And that is quite a difference.
Nevertheless, I think your suggestion is valid, as long as it is applied to the entire line, not on a per word basis.
